src/black/__init__.py

```diff
@@ -1078,6 +1078,7 @@ def f(
         for line in transform_line(
             current_line, mode=mode, features=split_line_features
         ):
+            line = fix_doublestar_in_op(line)
```

Comment:
I'm a little concerned by the performance impact this will have. Doing some rough testing, formatting `src/black/__init__.py` spends ~6500ms in the `format_str` method and ~4% (~300ms) of the total runtime was spent in the `fix_doublestar_in_op`... which is IMO too much for such a simple case like this. 

<details>

<summary>Profiling data (warning: extremely large image)</summary>

![out](https://user-images.githubusercontent.com/63936253/114799184-08cf6780-9d65-11eb-8c1a-7b437e1ee83d.png)

</details>

Is it possible to make this more efficient? Cloning the leaf seems to be where most of 300ms cost is coming from. Perhaps moving the logic somewhere into one of the transformers called in `transform_line` could help since some lines might not need this fixing? I'm sorry for my unadvice, I usually don't review formatting code.

Reply:
Ohh okay, gonna take a look. Thanks for pointing out!

Reply:
Agreed with @ichard26, it would be cleaner if it were a StringTransformer like the other transformation. This would additionally solve the performance issue. Few enough lines in Python have DOUBLESTAR tokens that the simplest way to improve performance is just look whether there even is doublestar on the line in `do_match()`.